runtime/connectors/gcs/gcs.go

```diff
@@ -103,3 +114,17 @@ func (c connector) ConsumeAsIterator(ctx context.Context, env *connectors.Env, s
 	}
 	return rillblob.NewIterator(ctx, bucketObj, opts)
 }
+
+func resolvedCredentials(ctx context.Context, env *connectors.Env) (*google.Credentials, error) {
+	if secretJSON, ok := env.Variables["GCS_CREDENTIALS"]; ok {
+		// gcs_credentials is set, use credentials from json string provided by user
+		// should we check for non empty strings only and fallback to hostcredentials otherwise ??
+		return google.CredentialsFromJSON(ctx, []byte(secretJSON), "https://www.googleapis.com/auth/cloud-platform")
+	}
+	// gcs_credentials is not set
+	if env.AllowHostCredentials {
+		// use host credentials
+		return gcp.DefaultCredentials(ctx)
+	}
+	return nil, fmt.Errorf("empty credentials: set gcs_credentials env variable")
```

Comment:
If possible, we should support no credentials – e.g. for public buckets. Is there something like anon credentials we can return here?

Reply:
There is no anonymous credentials in gcs but there is a anonymous client.